launch/test/launch/substitutions/test_command.py

```diff
@@ -29,14 +29,14 @@ def commands():
     this_dir = pathlib.Path(__file__).parent
 
     commands = {
-        'normal': str(this_dir / 'test_command' / 'normal_command.bash'),
-        'failing': str(this_dir / 'test_command' / 'failing_command.bash'),
-        'with_stderr': str(this_dir / 'test_command' / 'command_with_stderr.bash')
+        'normal': this_dir / 'test_command' / 'normal_command.bash',
+        'failing': this_dir / 'test_command' / 'failing_command.bash',
+        'with_stderr': this_dir / 'test_command' / 'command_with_stderr.bash'
     }
 
     if os.name == 'nt':
         for key, value in commands.items():
-            commands[key] = value.replace('bash', 'bat')
+            commands[key] = value.with_suffix('bat')
```

Comment:
I think this should be `'.bat'`

```suggestion
            commands[key] = value.with_suffix('.bat')
```

See the test failures, and:

```
$ python3
Python 3.10.12 (main, Feb  4 2025, 14:57:36) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pathlib import Path
>>> Path('/path/to/setup.bash').with_suffix('bat')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/pathlib.py", line 779, in with_suffix
    raise ValueError("Invalid suffix %r" % (suffix))
ValueError: Invalid suffix 'bat'
>>> Path('/path/to/setup.bash').with_suffix('.bat')
PosixPath('/path/to/setup.bat')
>>> 
```

Reply:
doh. thanks